socorro/external/postgresql/adi.py

```diff
@@ -0,0 +1,84 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import logging
+
+from socorro.external.postgresql.base import PostgreSQLBase
+from socorro.external import MissingArgumentError
+from socorro.lib import external_common
+
+logger = logging.getLogger("webapi")
+
+
+class ADI(PostgreSQLBase):
+
+    def get(self, **kwargs):
+        yesterday = datetime.datetime.utcnow() - datetime.timedelta(1)
+        tomorrow = yesterday + datetime.timedelta(2)
+        yesterday = yesterday.date()
+        tomorrow = tomorrow.date()
+        filters = [
+            ('start_date', yesterday, 'date'),
+            ('end_date', tomorrow, 'date'),
+            ('product', '', 'str'),
+            ('version', '', 'str'),
+        ]
+        params = external_common.parse_arguments(filters, kwargs)
+        required = (
+            'start_date',
+            'end_date',
+            'product',
+            'version',
+        )
+        missing = []
+        for each in required:
+            if not params.get(each):
+                missing.append(each)
+        if missing:
+            raise MissingArgumentError(', '.join(missing))
+
+        SQL = """
+        SELECT
+            SUM(adi_count) AS adi_count,
+            date::DATE,
+            product_name,
+            product_os_platform,
+            product_version,
+            update_channel
+        FROM
+            raw_adi
```

Comment:
Because this gets us closer to the "raw deal". My subtle goal is to stay away from too much normalization that is based on stored procedures. 
The `raw_adi` table isn't based on a stored procedure but on an insert (insert from select) statement that happens right after the `raw_adi_logs` table has been added to. In a cron job. 

Am I mad for doing this? Is it overzealous? 

Also, using adi_count directly avoids joins. I'm not sure if it makes a big difference.